src/Tribe/Body_Classes.php

```diff
@@ -0,0 +1,223 @@
+		 */
+		public function class_exists( $class ) {
+			return array_key_exists( $class, $this->classes );
+		}
+
+		/**
+		 * Checks if a class is in the queue and going to be added.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $class The class we are checking for.
+		 * @return boolean
+		 */
+		public function class_is_enqueued( $class ) {
+			if ( ! $this->class_exists( $class ) ) {
+				return false;
+			}
+
+			return $this->classes[$class];
+		}
+
+		/**
+		 * Dequeues a class.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $class
+		 * @return void|false
+		 */
+		public function dequeue_class( $class ) {
+			if ( ! $this->class_exists( $class ) ) {
+				return false;
+			}
+
+			$this->classes[$class] = false;
+		}
+
+		/**
+		 * Enqueues a class.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $class
+		 * @return void|false
+		 */
+		public function enqueue_class( $class ) {
+			if ( ! $this->class_exists( $class ) ) {
+				return false;
+			}
+
+			$this->classes[$class] = true;
+		}
+
+		/**
+		 * Add a single class to the queue.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $class The class to add.
+		 * @return void
+		 */
+		public function add_class( $class ) {
+			if ( empty( $class ) ) {
+				return;
+			}
+
+			if ( is_array( $class ) ) {
+				$this->add_classes( $class );
+			} else {
+				$class = sanitize_html_class( $class );
+				$this->classes[$class] = true ;
+			}
+		}
+
+		/**
+		 * Add an array of classes to the queue.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string> $class The classes to add.
+		 * @return void
+		 */
+		public function add_classes( array $classes ) {
+			$classes = array_map( 'sanitize_html_class', $classes );
+
+			foreach ( $classes as $key => $value ) {
+				// Just in case the classes are passed as class => bool, only add ones set to true.
+				if ( ! is_string( $value ) && false !== $value  ) {
+					$this->add_class( $key );
+				} else {
+					$this->add_class( $value );
+				}
+			}
+		}
+
+		/**
+		 * Remove a single class from the queue.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $class The class to remove.
+		 * @return void
+		 */
+		public function remove_class( $class ) {
+			$this->classes = array_filter(
+				$this->classes,
+				function( $v ) use ( $class ) {
+					return $v !== sanitize_html_class( $class );
+				}
+			);
+		}
+
+		/**
+		 * Remove an array of classes from the queue.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string> $classes The classes to remove.
+		 * @return void
+		 */
+		public function remove_classes( array $classes ) {
+			if ( empty( $classes ) || ! is_array( $classes) ) {
+				return;
+			}
+
+			foreach ( $classes as $class ) {
+				$this->remove_class( $class );
+			}
+		}
+
+		/**
+		 * Adds the enqueued classes to the body class array.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string> $classes An array of body class names.
+		 * @return void
+		 */
+		private function maybe_add_body_classes( $classes = [] ) {
```

Comment:
I personally do not like the `maybe_` in public methods or functions as it leaks information about the "how" and does not align with "tell, do not ask".
Personal take, though.

Reply:
I tend to agree here, it's easier to follow if code is not imperative but rather declarative.
Also, if `$this->classes` was an empty array, wouldn't doing `array_merge( $classes, $this->get_class_names() );` return the same result as just returning `$classes`?